test/any/datastore/rate_limited_DAO-test.es6.js

```diff
@@ -5,50 +5,68 @@
 
 describe('RateLimitedDAO', function() {
   let rateLimitedDAO;
-  foam.CLASS({
-    package: 'test.RateLimitedDAO',
-    name: 'item',
-    properties: ['id'],
-  });
   beforeEach(function() {
+    jasmine.clock().install();
     rateLimitedDAO = org.chromium.apis.web.RateLimitedDAO.create({
-      interval: 100,
+      interval: 50,
       delegate: foam.dao.ArrayDAO.create(),
     });
+    foam.CLASS({
+      package: 'test.RateLimitedDAO',
+      name: 'Item',
+      properties: ['id'],
+    });
+  });
+
+  afterEach(function() {
+    jasmine.clock().uninstall();
   });
+
   it('puts objects at specified interval.', function(done) {
-    rateLimitedDAO.put(test.RateLimitedDAO.item.create({id: 1}));
-    rateLimitedDAO.put(test.RateLimitedDAO.item.create({id: 2}));
-    rateLimitedDAO.put(test.RateLimitedDAO.item.create({id: 3}));
-    setTimeout(() => {
-      rateLimitedDAO.select().then((arraySink) => {
-        expect(arraySink.a.length).toBe(1);
-      });
-    }, 100);
-    setTimeout(() => {
-      rateLimitedDAO.select().then((arraySink) => {
-        expect(arraySink.a.length).toBe(2);
-      });
-    }, 200);
-    setTimeout(() => {
-      rateLimitedDAO.select().then((arraySink) => {
-        expect(arraySink.a.length).toBe(3);
-      });
-    }, 300);
-    setTimeout(() => {
-      done();
-    }, 500);
+    rateLimitedDAO.put(test.RateLimitedDAO.Item.create({id: 1}));
+    rateLimitedDAO.put(test.RateLimitedDAO.Item.create({id: 2}));
+    rateLimitedDAO.put(test.RateLimitedDAO.Item.create({id: 3}));
+    jasmine.clock().tick(55);
+    rateLimitedDAO.delegate.select().then((arraySink) => {
+      expect(arraySink.a.length).toBe(1);
+    });
+    jasmine.clock().tick(50);
+    rateLimitedDAO.delegate.select().then((arraySink) => {
+      expect(arraySink.a.length).toBe(2);
+    });
+    jasmine.clock().tick(50);
+    rateLimitedDAO.delegate.select().then((arraySink) => {
+      expect(arraySink.a.length).toBe(3);
+    });
+    done();
   });
-  it('puts items into its delegate DAO.', function(done) {
-    rateLimitedDAO.put(test.RateLimitedDAO.item.create({id: 1}));
-    rateLimitedDAO.put(test.RateLimitedDAO.item.create({id: 2}));
-    rateLimitedDAO.put(test.RateLimitedDAO.item.create({id: 3}));
-    setTimeout(() => {
-      rateLimitedDAO.select().then((arraySink) => {
-        expect(arraySink.a.map((item) => item.id).sort()).toEqual(
-          [1, 2, 3]);
-          done();
-      });
-    }, 500);
+  it('sucessfully puts Items into its delegate DAO.', function(done) {
+    rateLimitedDAO.put(test.RateLimitedDAO.Item.create({id: 1}));
+    rateLimitedDAO.put(test.RateLimitedDAO.Item.create({id: 2}));
+    rateLimitedDAO.put(test.RateLimitedDAO.Item.create({id: 3}));
+    jasmine.clock().tick(160);
+    rateLimitedDAO.delegate.select().then((arraySink) => {
```

Comment:
We should check that `put()`s were issued to delegate in order. I think it's fine to,  along with this check or instead of it, peek at `delegate` `ArrayDAO`'s array and check for `[1, 2, 3]` *without* sorting. The *thorough* way to test would be to create a custom DAO in `beforeEach()` -- basically a clone of `ArrayDAO`; then, either the custom DAO checks that "id being `put()`" = "last id `put()` + 1" or just peek at the array after operations are complete (as suggested as a less thorough solution).